packages/platforms/accton/x86-64/wedge100bf-32x/onlp/builds/x86_64_accton_wedge100bf_32x/module/src/platform_lib.c

```diff
@@ -312,7 +312,19 @@ bmc_file_read_int(int* value, char *file, int base)
 }
 
 int
-bmc_i2c_readb(uint8_t bus, uint8_t devaddr, uint8_t addr)
+bmc_i2c_readb(uint8_t bus, uint8_t devaddr)
+{
+    int ret = 0, value;
+    char cmd[64] = {0};
+
+    snprintf(cmd, sizeof(cmd), "i2cget -f -y %d 0x%x\r\n", bus, devaddr);
+    ret = bmc_command_read_int(&value, cmd, 16);
+    return (ret < 0) ? ret : value;
```

Comment:
Can’t this function call `bmc_i2c_readbdata(bus, devaddr, NULL)` to avoid duplication?

Reply:
It can't. 3rd parameter is a uint8_t.

Reply:
Sorry, then `bmc_i2c_readbdata(bus, devaddr, 0)`.

Reply:
I suppose that It couldn't use addr=0 to call original bmc_i2c_readb().
Because caller may need to read from addr 0 by  bmc_i2c_readbdata().

Reply:
Then make it signed, and use -1 for no address. ;-)

I see, it’s more complicated than I thought. Sorry for the noise.